src/pages/home/report/ReportActionItem.tsx

```diff
@@ -181,909 +31,57 @@ function ReportActionItem({
-                                            displayNamesWithTooltips={displayNamesWithTooltips}
-                                            tooltipEnabled
-                                            numberOfLines={1}
-                                            textStyles={[styles.chatItemMessageHeaderTimestamp, styles.flex1]}
-                                            shouldUseFullTitle={isWhisperOnlyVisibleByUser}
-                                        />
-                                    </View>
-                                )}
-                                {renderReportActionItem(!!hovered || !!isReportActionLinked, isWhisper, hasErrors)}
-                            </OfflineWithFeedback>
-                        </View>
-                    </View>
-                )}
-            </Hoverable>
-            <View style={styles.reportActionSystemMessageContainer}>
-                <InlineSystemMessage message={action.error} />
-            </View>
-        </PressableWithSecondaryInteraction>
+        <PureReportActionItem
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...props}
+            action={action}
+            report={report}
+            draftMessage={draftMessage}
+            iouReport={iouReport}
+            emojiReactions={emojiReactions}
+            linkedTransactionRouteError={linkedTransactionRouteError}
+            reportNameValuePairs={reportNameValuePairs}
+            isUserValidated={isUserValidated}
+            parentReport={parentReport}
+            personalDetails={personalDetails}
+            blockedFromConcierge={blockedFromConcierge}
+            originalReportID={originalReportID}
+            deleteReportActionDraft={Report.deleteReportActionDraft}
+            isArchivedRoom={ReportUtils.isArchivedRoomWithID(originalReportID)}
+            isChronosReport={ReportUtils.chatIncludesChronosWithID(originalReportID)}
+            toggleEmojiReaction={Report.toggleEmojiReaction}
+            createDraftTransactionAndNavigateToParticipantSelector={ReportUtils.createDraftTransactionAndNavigateToParticipantSelector}
+            resolveActionableReportMentionWhisper={Report.resolveActionableReportMentionWhisper}
+            resolveActionableMentionWhisper={Report.resolveActionableMentionWhisper}
+            isClosedExpenseReportWithNoExpenses={ReportUtils.isClosedExpenseReportWithNoExpenses(iouReport)}
+            isCurrentUserTheOnlyParticipant={ReportUtils.isCurrentUserTheOnlyParticipant}
+            missingPaymentMethod={missingPaymentMethod}
+            reimbursementDeQueuedActionMessage={ReportUtils.getReimbursementDeQueuedActionMessage(
+                action as OnyxEntry<ReportAction<typeof CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENT_DEQUEUED>>,
+                report,
+            )}
+            forReportAction={ModifiedExpenseMessage.getForReportAction(reportID, action)}
```

Comment:
```suggestion
            getForReportAction={ModifiedExpenseMessage.getForReportAction(reportID, action)}
```

Reply:
We'll return string here are you sure with the naming?

Reply:
TBH that's a terrible function name 😂 I think in that case we should rename it to `modifiedExpenseMessage` since that's what the function is returning